akka-docs-dev/rst/java/code/docs/http/javadsl/HandlerExampleSpec.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package docs.http.javadsl;
+
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.server.*;
+import akka.http.javadsl.server.values.Parameters;
+import akka.http.javadsl.server.values.PathMatchers;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+import akka.http.javadsl.testkit.TestRoute;
+import org.junit.Test;
+
+public class HandlerExampleSpec extends JUnitRouteTest {
+    @Test
+    public void testSimpleHandler() {
+        //#simple-handler-example-full
+        class TestHandler extends akka.http.javadsl.server.AllDirectives {
+            //#simple-handler
+            Handler handler = new Handler() {
```

Comment:
Yeah, for many SAM interfaces now we could move to `akka.japi.function` since we have them (we did not previously have them for so many params).

I thought of this direction https://github.com/ktoso/akka/commit/8784c18f24e46e34c4138dbd8982a96ccdf12c2a#diff-58791b179be1c95f3090b06bb72bf473R1 but then we need to name then `handleWithN` since javac is not-very-good-with-overloads: https://gist.github.com/ktoso/76248c629bd614c81624

If we change then to extend the matching function interfaces and have function in method params as well, why keep the `Handler2`? If users want to they can make their Handler, and if we don't require a type in APIs, let's not have it.

Maybe I'm overthinking, but I feel that the API proposed in the above commit I linked "looks more like akka streams", so maybe javadsl users would be more familiar with it... The `handleWith##` uglieness is a cost though :~

Do you have an opinion on this @rkuhn @viktorklang (using Roland while it's still monday ;-))